sklearn/utils/estimator_checks.py

```diff
@@ -1521,22 +1521,30 @@ def check_estimators_overwrite_params(name, estimator_orig):
 
 
 @ignore_warnings(category=(DeprecationWarning, FutureWarning))
-def check_no_fit_attributes_set_in_init(name, Estimator):
-    """Check that Estimator.__init__ doesn't set trailing-_ attributes."""
-    # this check works on classes, not instances
-    estimator = Estimator()
-    for attr in dir(estimator):
-        if attr.endswith("_") and not attr.startswith("__"):
-            # This check is for properties, they can be listed in dir
-            # while at the same time have hasattr return False as long
-            # as the property getter raises an AttributeError
-            assert_false(
-                hasattr(estimator, attr),
-                "By convention, attributes ending with '_' are "
-                'estimated from data in scikit-learn. Consequently they '
-                'should not be initialized in the constructor of an '
-                'estimator but in the fit method. Attribute {!r} '
-                'was found in estimator {}'.format(attr, name))
+def check_no_attributes_set_in_init(name, estimator):
+    """Check setting during init. """
+
+    if hasattr(type(estimator).__init__, "deprecated_original"):
+        return
+
+    init_params = _get_args(type(estimator).__init__)
+    base_params = (_get_parent_args(type(estimator)) +
+                   [a for a in dir(type(estimator))
+                    if not a.startswith("__")])
+
+    # Test for no setting apart from parameters during init
+    for attr, val in vars(estimator).items():
```

Comment:
Very minor issue: This will raise a TypeError for an estimator without `__dict__`. I've never been sure if we want to force estimators to have `__dict__`. I don't see why you'd ever need an estimator to not have a `__dict__`... For all I know we have other common tests which have this implicit requirement, but since it is the default for Python objects to have a `__dict__`, we never make it explicit...

Reply:
PS: I think this is fine. Don't change it.